nntrainer/context.h

```diff
@@ -36,6 +36,19 @@
 
 namespace nntrainer {
 
+/**
+ * @class ContextData contains Data which generated by Context to use in Layers
+ * created by Context
+ * @brief Container to hold context data generated by Context. Eventaully
+ * RunLayerContext in layer_node will hold the ContextData, so that, Layer can
+ * access this Context Data.
+ */
+class ContextData {
```

Comment:
We might want publicly inherit from `std::enable_shared_from_this<ContextData>`, since it will be used with `std::shared_pointer`. Upside would be that shared/weak atomic reference counters (shared_ptr control block) will become part of this object rather than allocated (most likely dynamic allocation).